File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        int answer = 1;

		for (String[] clothe : clothes) {
			String key = clothe[1];
			map.put(key, map.getOrDefault(key, 0) + 1);
		}
        for (Map.Entry<String, Integer> entry : map.entrySet()) {
            answer *= entry.getValue() + 1;
        }
        return answer - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.kt
class Solution {
    fun solution(clothes: Array<Array<String>>): Int {
        var answer = 1
        
        val clothesMap = mutableMapOf<String, Int>()
        
        for (cloth in clothes) {
            val type = cloth[1]
            clothesMap[type] = clothesMap.getOrDefault(type, 0) + 1
        }
        
        for (count in clothesMap.values) {
            answer *= (count+1)
        }
        
        return answer - 1
    }
}